cfme/fixtures/v2v.py

```diff
@@ -370,119 +328,106 @@ def form_data_vm_obj_dual_nics(request, appliance, source_provider, provider):
     )
     vm_obj = get_vm(request, appliance, source_provider, request.param[2], request.param[0])
-    return FormDataVmObj(form_data=form_data, vm_list=[vm_obj])
+    return FormDataVmObj(infra_mapping_data=infra_mapping_data, vm_list=[vm_obj])
 
 
 @pytest.fixture(scope="function")
 def form_data_vm_obj_single_network(request, appliance, source_provider, provider):
-    form_data = _form_data(source_provider, provider)
+    infra_mapping_data = infra_mapping_default_data(source_provider, provider)
     recursive_update(
-        form_data,
+        infra_mapping_data,
         {
-            "general": {
-                "description": "Single Network migration of VM from {vlan1} to {vlan2},".format(
-                    vlan1=request.param[0], vlan2=request.param[1]
+            "description": "Single Network migration of VM from {vlan1} to {vlan2},".format(
+                vlan1=request.param[0], vlan2=request.param[1]
+            ),
+            "networks": [
+                component_generator(
+                    "vlans", source_provider, provider, request.param[0], request.param[1]
                 )
-            },
-            "network": {
-                "Cluster ({})".format(provider.data.get("clusters")[0]): {
-                    "mappings": [
-                        _form_data_mapping(
-                            "vlans", source_provider, provider, request.param[0], request.param[1]
-                        )
-                    ]
-                }
-            },
+            ],
         },
     )
     vm_obj = get_vm(request, appliance, source_provider, request.param[2])
-    return FormDataVmObj(form_data=form_data, vm_list=[vm_obj])
+    return FormDataVmObj(infra_mapping_data=infra_mapping_data, vm_list=[vm_obj])
 
 
-def _form_data_mapping(selector, source_provider, provider, source_list=None, target_list=None):
+def component_generator(selector, source_provider, provider, source_list=None, target_list=None):
+    """
+    Component generator method to generate a dict of source and target
+    components(cluster/datastore/networks).
+
+    Args:
+        selector: can be cluster/datastore/network
+        source_provider: vmware provider to migrate from
+        provider:  rhev or osp provider or target provider to migrate to
+        source_list: source cluster/datastore/network on vmware provider to migrate from
+        target_list: target cluster/ds/network to migrate to
+
+    """
     source_data = source_provider.data.get(selector, [])
     target_data = provider.data.get(selector, [])
+
     if selector is "clusters":
-        source = source_data if source_data else None
-        target = target_data if target_data else None
-        skip_test = not source or not target
+        sources = source_data if source_data else None
+        targets = target_data if target_data else None
+        component = InfraMapping.ClusterComponent(
+            [partial_match(sources[0])], [partial_match(targets[0])]
```

Comment:
This will cause a TypeError if `sources` was set to `None` above.

Logic needs to be reworked a bit here to make this flow a bit better.

There are a few options:
1. Check `source_data` and `target_data` at ~L410 and skip immediately if they're empty
2. Set component to `None` if `sources` or `targets` is empty, modify `skip_test` setting on L430 to check if component is `None`